Functions/BIDS/xASL_bids_GenerateParticipantsJSON.m

```diff
@@ -0,0 +1,152 @@
+function xASL_bids_GenerateParticipantsJSON(x)
+%xASL_bids_GenerateParticipantsJSON This function generates the participants.json sidecar to the participants.tsv
+%
+% FORMAT: xASL_bids_GenerateParticipantsJSON(x)
+%
+% INPUT:
+%   x       - x structure containing all input parameters (REQUIRED)
+%   participants.tsv needs to be present
+%
+% OUTPUT: n/a 
+%   participants.json will be written
+% -----------------------------------------------------------------------------------------------------------------------------------------------------
+% DESCRIPTION: This function generates the participants.json sidecar to the participants.tsv, using the following steps
+%
+% 1. Load pre-existing participants.json
+% 2. Parse participants.tsv
+% 3. Parse standard keys
+% 4. Parse ASL-related keys
+% 4a. Replace key abbreviations with verbose description
+% 4b. Manage units
+% 5. (over-)Write the new json
+%
+% EXAMPLE: xASL_bids_GenerateParticipantsJSON(x);
+% -----------------------------------------------------------------------------------------------------------------------------------------------------
+% Copyright 2015-2025 ExploreASL
+% Licensed under Apache 2.0, see permissions and limitations at
+% https://github.com/ExploreASL/ExploreASL/blob/main/LICENSE
+% you may only use this file in compliance with the License.
+% __________________________________
+
+
+
+%% 1. Load pre-existing participants.json
+pathJson = fullfile(x.dir.xASLDerivatives, 'participants.json');
+pathTSV = fullfile(x.dir.xASLDerivatives, 'participants.tsv');
+if exist(pathJson, 'file')
+    % Load the JSON, but note that we will overwrite it if fields already exist
+    jsonParticipants = xASL_io_ReadJson(pathJson);
+end
+
+
+%% 2. Parse participants.tsv
+jsonParticipants = struct();
```

Comment:
this will remove anything read above. You need to add it to "else" above.

Reply:
Done in a882a8bbda1b10a30e0d3bbadb30ba880788f4ad